README.md

```diff
@@ -1,7 +1,111 @@
-### ReactiveFeedback
+# ReactiveFeedback
 
 Unideraction reactive architecture. This is [ReactiveSwift](https://github.com/ReactiveCocoa/ReactiveSwift) implemetation of the [RxFeedback](https://github.com/kzaher/RxFeedback)
 
-### Documentation
+## Documentation
 
-* TBD
+![](diagrams/ReactiveFeedback.jpg)
+
+### Motivation
+
+Requirements for iOS apps have become huge. Our code has to manage a lot of state e.g server responses, cached data, UI state, routing etc. Some may say that Reactive Programming can help us a lot, but in wrong hands, it can make even more harm to your code base.
+
+The goal of this library to provide a simple and intuitive approach to design reactive state machines.
+
+### Core Concepts
+
+##### State 
+
+Is a single source of truth. It represents a state of your system. Usually, plane swift type (does not contain any ReactiveSwift primitives) to the point that it can be saved on disk. The only way to change a `State` is to emit an `Event`
```

Comment:
`State` is `the` single source of truth.

Reply:
~plane~ `a plain` swift type (`which` does not contain any ReactiveSwift primitives)